drivers/wifi/nrf700x/zephyr/src/zephyr_net_if.c

```diff
@@ -106,6 +196,27 @@ void wifi_nrf_if_init(struct net_if *iface)
 
 	net_if_set_link_addr(iface, (unsigned char *)&rpu_ctx_zep->mac_addr,
 			     sizeof(rpu_ctx_zep->mac_addr), NET_LINK_ETHERNET);
+
+
+	net_mgmt_init_event_callback(&ip_maddr4_add_cb,
+				     ip_maddr_event_handler,
+				     NET_EVENT_IPV4_MADDR_ADD);
+	net_mgmt_add_event_callback(&ip_maddr4_add_cb);
+
+	net_mgmt_init_event_callback(&ip_maddr4_del_cb,
+				     ip_maddr_event_handler,
+				     NET_EVENT_IPV4_MADDR_DEL);
+	net_mgmt_add_event_callback(&ip_maddr4_del_cb);
+
+	net_mgmt_init_event_callback(&ip_maddr6_add_cb,
+				     ip_maddr_event_handler,
+				     NET_EVENT_IPV6_MADDR_ADD);
+	net_mgmt_add_event_callback(&ip_maddr6_add_cb);
+
+	net_mgmt_init_event_callback(&ip_maddr6_del_cb,
+				     ip_maddr_event_handler,
+				     NET_EVENT_IPV6_MADDR_DEL);
+	net_mgmt_add_event_callback(&ip_maddr6_del_cb);
```

Comment:
You can add them in a single CB handler, by ORing the events 

```
net_mgmt_init_event_callback(&ip_madd_cbs,
                  ip_maddr_event_handler, 
                  NET_EVENT_IPV4_MADDR_ADD |  NET_EVENT_IPV4_MADDR_DEL | 
                   NET_EVENT_IPV6_MADDR_ADD | NET_EVENT_IPV6_MADDR_DEL);
```

Reply:
As discussed offline, the above won't work as we are trying to mix two layers of net_mgmt IPv4 andIPv6, the check in `mgmt_run_callbacks` fails, so, divide in to two handlers ipv4 and ipv6 and you can use multiple events.